src/test/ui/regions/regions-assoc-type-static-bound-in-trait-not-met.stderr

```diff
@@ -1,15 +1,16 @@
-error[E0477]: the type `&'a i32` does not fulfill the required lifetime
+error[E0478]: lifetime bound not satisfied
   --> $DIR/regions-assoc-type-static-bound-in-trait-not-met.rs:10:18
    |
 LL |     type Value = &'a i32;
    |                  ^^^^^^^
    |
-note: type must satisfy the static lifetime as required by this binding
-  --> $DIR/regions-assoc-type-static-bound-in-trait-not-met.rs:5:17
+note: lifetime parameter instantiated with the lifetime `'a` as defined here
+  --> $DIR/regions-assoc-type-static-bound-in-trait-not-met.rs:9:6
    |
-LL |     type Value: 'static;
-   |                 ^^^^^^^
+LL | impl<'a> Foo for &'a i32 {
+   |      ^^
+   = note: but lifetime parameter must outlive the static lifetime
 
 error: aborting due to previous error
 
-For more information about this error, try `rustc --explain E0477`.
+For more information about this error, try `rustc --explain E0478`.
```

Comment:
I think this PR could also fix `E0478` to not show `must outlive the static lifetime` error message. See #41816.